core/common/src/main/java/alluxio/PropertyKey.java

```diff
@@ -2406,6 +2406,13 @@ public String toString() {
           .setConsistencyCheckLevel(ConsistencyCheckLevel.WARN)
           .setScope(Scope.CLIENT)
           .build();
+  public static final PropertyKey USER_APP_ID =
+      new Builder(Name.USER_APP_ID)
```

Comment:
Keep in mind that multiple clients sharing the same configuration file would have the same app id, which would defeat the purpose of having a client id.

Reply:
Right, if the user sets the app id manually, they should make sure they set it uniquely per logical application (for example in a Spark job).

Reply:
Right. Can you call this out explicitly? Other than it, this PR looks good to me.

Reply:
Updated, thanks.